lib/kdtree.h

```diff
@@ -66,6 +67,9 @@ class FlatNode {
     constexpr static uint32_t INVALID_TRIANGLE_ID = 0xFFFFFFFF >> 2;
 
 public:
+
+    FlatNode() {};
```

Comment:
This is one drawback of Cereal. It requires a default constructor. It can go without but then we have to handle pointers. See *Default construction* section in the [docs](http://uscilab.github.io/cereal/serialization_functions.html).